ui/src/app/components/routing.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+
+import {withRoute} from 'app/routes';
+import {history} from 'app/services/history';
+
+export const Router = withRoute()(({route}) => {
+  if (!route) {
+    return <div>Not found</div>;
```

Comment:
This is where the 'not found' page is rendered. Alternatively, we could add a catch-all route at the end of the routes list, and handle it that way.